crates/weechat/src/config/enum.rs

```diff
@@ -0,0 +1,168 @@
+use std::marker::PhantomData;
+
+use weechat_sys::{t_config_option, t_weechat_plugin};
+
+use crate::{
+    config::{
+        config_options::{ConfigOptions, FromPtrs, HiddenConfigOptionT},
+        BaseConfigOption, ConfigSection,
+    },
+    Weechat,
+};
+
+/// Settings for a new string option.
+#[derive(Default)]
+pub struct EnumOptionSettings {
+    pub(crate) name: String,
+
+    pub(crate) description: String,
+
+    pub(crate) default_value: i32,
+
+    pub(crate) min: i32,
+
+    pub(crate) max: i32,
+
+    pub(crate) string_values: String,
+
+    pub(crate) change_cb: Option<Box<dyn FnMut(&Weechat, &EnumOption)>>,
+}
+
+impl EnumOptionSettings {
+    /// Create new settings that can be used to create a new string option.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name of the new option.
+    pub fn new<N: Into<String>>(name: N) -> Self {
+        EnumOptionSettings { name: name.into(), ..Default::default() }
+    }
+
+    /// Set the description of the option.
+    ///
+    /// # Arguments
+    ///
+    /// * `description` - The description of the new option.
+    pub fn description<D: Into<String>>(mut self, description: D) -> Self {
+        self.description = description.into();
+        self
+    }
+
+    /// Set the default value of the option.
+    ///
+    /// This is the value the option will have if it isn't set by the user. If
+    /// the option is reset, the option will take this value.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - The value that should act as the default value.
+    pub fn default_value<V: Into<i32>>(mut self, value: V) -> Self {
+        self.default_value = value.into();
+        self
+    }
+
+    /// Set minimal value of the integer option.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - The values that should act as minimal valid value.
+    pub fn min(mut self, value: i32) -> Self {
+        self.min = value;
+        self
+    }
+
+    /// Set maximum value of the integer option.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - The values that should act as maximal valid value.
+    pub fn max(mut self, value: i32) -> Self {
+        self.max = value;
+        self
+    }
+
+    /// Set the string values of the option.
+    ///
+    /// This setting decides if the integer option should act as an enum taking
+    /// symbolic values.
+    ///
+    /// # Arguments
+    ///
+    /// * `values` - The values that should act as the symbolic values.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// use weechat::config::IntegerOptionSettings;
```

Comment:
There's a copy/paste error here, the docs are talking about `IntegerOptionSettings`.

Reply:
Fixed, alongside with even more copy-paste error references to string/int options + added small comment about how enum behaves (in `new()` for `EnumOptionSettings`)